src/services/item/plugins/published/repositories/itemPublished.ts

```diff
@@ -36,6 +42,40 @@ export const ItemPublishedRepository = AppDataSource.getRepository(ItemPublished
     });
   },
 
+  // return public item entry? contains when it was published
+  async getAllItems() {
+    const publishedRows = await this.find({ relations: { item: true } });
+    return publishedRows.map(({ item }) => item);
+  },
+
+  // Must Implement a proper Paginated<Type> if more complex pagination is needed in the future
+  async getPaginatedItems(
+    page: number = 1,
+    pageSize: number = 20,
+  ): Promise<[ItemPublished[], number]> {
+    const [items, total] = await this.findAndCount({
+      relations: { item: { creator: true } },
+      take: pageSize,
+      skip: (page - 1) * pageSize,
+    });
+    return [items, total];
+  },
+
+  // return public item entry? contains when it was published
+  async getItemsForMember(memberId: string) {
+    // get for membership write and admin -> createquerybuilder
+    return this.createQueryBuilder()
+      .select(['item'])
+      .from(Item, 'item')
```

Comment:
I think I did something similar and it resulted in the huge queries we removed. Maybe check it out.

Reply:
Looks like it's a merge error during my rebase, it's actually one of the method you fixed and move to ItemRepository